be/src/pipeline/exec/scan_operator.cpp

```diff
@@ -1456,36 +1450,19 @@ Status ScanOperatorX<LocalStateType>::get_block(RuntimeState* state, vectorized:
     }};
 
     if (state->is_cancelled()) {
-        // ISSUE: https://github.com/apache/doris/issues/16360
-        // _scanner_ctx may be null here, see: `VScanNode::alloc_resource` (_eos == null)
-        if (local_state._scanner_ctx) {
-            local_state._scanner_ctx->set_status_on_error(Status::Cancelled("query cancelled"));
-            return local_state._scanner_ctx->status();
-        } else {
-            return Status::Cancelled("query cancelled");
-        }
+        return Status::Cancelled("query cancelled");
```

Comment:
I think you should call scanner_context.set status on error, to cancel the scanner as soon as possible.